### PR TITLE
Tiny js world task

### PR DESCRIPTION
**Pull request by KikinovK**
_Complete the form below and remove this line._

### Tiny js world task

[Demo](https://kikinovk.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/KikinovK/frontend-2021-homeworks/tree/tiny-js-world-task/submissions/kikinovk/Tiny%20JS%20World)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/kikinovk/Tiny JS World/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+};
+dog.friends = [man, woman];
+cat.friends = [man, woman, catWoman];
+man.friends = [woman, dog, cat, catWoman];
+woman.friends = [dog, cat, man];
+catWoman.friends = [cat, man];
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+
+const Description = obj => {
```
Comment (OleksiyRudenko) [inline submissions/kikinovk/Tiny JS World/index.js:58]: This is not a class but rather a regular function. We do not need capitalization in its name.

File: submissions/kikinovk/Tiny JS World/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const nameFriends = obj.friends.reduce((result, item) => result + `${item.name}; `, '')
+  return `${obj.species}; ${obj.name}; ${obj.gender}; ${obj.legs}; ${obj.hands}; ${obj.saying}; ${nameFriends}`
```
Comment (OleksiyRudenko) [inline submissions/kikinovk/Tiny JS World/index.js:60]: There is a pattern here. Can you make code [DRYer](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and cleaner by using a list of key/property names, mapping it into values and joining with a delimiter?
Array methods are to your rescue.

File: submissions/kikinovk/Tiny JS World/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+dog.friends = [man, woman];
+cat.friends = [man, woman, catWoman];
+man.friends = [woman, dog, cat, catWoman];
+woman.friends = [dog, cat, man];
+catWoman.friends = [cat, man];
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+
+const Description = obj => {
+  const nameFriends = obj.friends.reduce((result, item) => result + `${item.name}; `, '')
```
Comment (OleksiyRudenko) [inline submissions/kikinovk/Tiny JS World/index.js:59]: How about `.map` and then `.join`? `reduce` is universal but yet too generic.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@KikinovK good job!
A few improvements to make it great :)
Think big. E.g. what if you have 20 properties per object?

**Comment by OleksiyRudenko** [comment]
@KikinovK does your code work? Code refactoring may require more efforts.
Please update demo.

**Comment by KikinovK** [comment]
code works, already demo updated.

**Review by OleksiyRudenko — APPROVED** [review]
@KikinovK Well done!

**Pull request merged**